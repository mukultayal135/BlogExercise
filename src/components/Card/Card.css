.card {
  height: 450px;
  width: 300px;
  display: flex;
  flex-direction: column;
  border: 2px solid lightgray;
}
