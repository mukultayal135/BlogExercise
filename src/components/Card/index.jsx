import PropTypes from 'prop-types';
import { useState } from 'react';
import Img from '../Img';
import PostContent from '../PostContent';
import Reactions from '../Reactions';
import './Card.css';

function Card({ img, title, description, readingTime, date, initialClaps }) {
  const [claps, setClaps] = useState(initialClaps);
  const [like, setLike] = useState(false);
  const onReact = () => {
    setClaps(claps + 1);
  };
  const onLike = () => {
    setLike(!like);
  };
  return (
    <div className="card" data-testid="card">
      <Img path={img} />
      <PostContent
        title={title}
        description={description}
        readingTime={readingTime}
        date={date}
      />
      <Reactions onReact={onReact} count={claps} onLike={onLike} liked={like} />
    </div>
  );
}
Card.propTypes = {
  img: PropTypes.string.isRequired,
  initialClaps: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  readingTime: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};
export default Card;
