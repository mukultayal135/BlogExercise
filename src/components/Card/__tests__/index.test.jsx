/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { fireEvent, screen, render } from '@testing-library/react';
import Card from '..';

describe('Card', () => {
  const mockProps = {
    date: '2nd Januray, 2018',
    readingTime: '2 mins',
    title: 'The future of abstract art and the culture ...',
    description:
      'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...',
    initialClaps: 10,
    img: 'abstract.png',
  };
  it('should render Card component', () => {
    const { asFragment } = render(<Card {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should increase clap count by 1 when click on clap icon', () => {
    render(<Card {...mockProps} />);
    const clap = screen.getByTestId('count-clap');
    const initialCount = clap.textContent;
    expect(initialCount).toBe('10');
    const clapIcon = screen.getByAltText('clapImage');
    fireEvent.click(clapIcon);
    expect(clap.textContent).toBe('11');
  });
  it('should change like icon when click on like icon', () => {
    render(<Card {...mockProps} />);
    const likeIcon = screen.getByAltText('heartImg');
    expect(likeIcon.src).toContain('heart-black.svg');
    fireEvent.click(likeIcon);
    expect(likeIcon.src).toContain('heart-red.svg');
  });
});
