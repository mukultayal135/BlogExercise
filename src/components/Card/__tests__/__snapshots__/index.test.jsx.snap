// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render Card component 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="card"
  >
    <div>
      <img
        alt=""
        class="card-image"
        src="abstract.png"
      />
    </div>
    <div
      class="card-content card-padding"
    >
      <div
        class="date-time"
      >
        <div
          class="date"
        >
          2nd Januray, 2018
        </div>
        <div
          class="read-time"
        >
          2 mins
        </div>
      </div>
      <div
        class="card-heading"
      >
        The future of abstract art and the culture ...
      </div>
      <div
        class="card-description"
      >
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </div>
    </div>
    <hr />
    <div
      class="reactions"
    >
      <div
        class="clap"
      >
        <img
          alt="clapImage"
          src="clapping.svg"
        />
        <div
          class="count"
          data-testid="count-clap"
        >
          10
        </div>
      </div>
      <div
        class="heart"
      >
        <img
          alt="heartImg"
          src="heart-black.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
